message-hub/README.md

```diff
@@ -0,0 +1,174 @@
+topics=exampleTopic
+# Substitute with your DB
+cloudant.db.url=https://examples.cloudant.com/animaldb
+# Not needed for the public examples account, but if using your own data then
+# substitute with your Cloudant credentials
+#cloudant.db.username=
+#cloudant.db.password=
+cloudant.db.since=0
+batch.size=50
+# Configure kafka-connect-cloudant to generate structs from the JSON documents
+cloudant.value.schema.struct=true
+# Flattening is not needed for this tutorial, but if you have nested objects and
+# arrays in your JSON documents consider enabling flattening for use with the
+# kafka-connect-jdbc sink. This may not be necessary for other sink connectors
+# that can handle nested structs.
+cloudant.value.schema.struct.flatten=false
+# Configure kafka-connect-cloudant to omit design documents
+cloudant.omit.design.docs=true
+```
+
+# Configuration for consuming messages from Message Hub and writing to Db2 Warehouse on Cloud using Apache Kafka's integrated jdbc-sink-connector
+
+1. Download the [`kafka-connect-jdbc` jar from Confluent's Maven repository](packages.confluent.io/maven/io/confluent/kafka-connect-jdbc/4.0.0/kafka-connect-jdbc-4.0.0.jar)
+1. Create a `connect-jdbc-sink.properties` file
+1. Add these properties, substituting your details where appropriate.
+
+```properties
+# Substitute with a name for your sink connector instance
+name=db2-kafka-example
+connector.class=io.confluent.connect.jdbc.JdbcSinkConnector
+# Substitute with the name of your topic
+topics=exampleTopic
+# Substitute with the JDBC connection URL from your Db2 Warehouse on Cloud copied earlier
+connection.url=jdbc:db2://dashdb-entry-yp-lon02-01.services.eu-gb.bluemix.net:50001/BLUDB:user=username;password=password;sslConnection=true;
+batch.size=1
+table.name.format=ANIMALS
+```
+
+Note that these options are not currently supported by the JDBC sink connector when using the `db2` JDBC dialect:
+
+* `auto.create` for creating SQL tables
+* `auto.evolve` for adjusting SQL tables
+* `insert.mode=upsert` or `insert.mode=update`
+
+# Running Apache Kafka Connect to produce and consume messages
+
+1. Add the kafka-connect-cloudant connector, the kafka-connect-jdbc connector,
+and the Db2 JDBC 4 jar (installed as part of the Db2 driver package) to the Java
+`CLASSPATH` as appropriate for your platform (note this example uses the default
+`sqllib/java` install path) e.g.
+```sh
+export CLASSPATH=/path/to/downloaded/kafka-connect-cloudant-0.100.0-kafka-1.0.0.jar:/path/to/downloaded/kafka-connect-jdbc-4.0.0.jar:/sqllib/java/db2jcc4.jar
+```
+Note this example uses the Linux default installation location for the db2jcc4.jar it may be diff
+1. Run the Kafka Connect standalone worker with the property files created and configured earlier.
+```sh
+/path/to/your//kafka_2.11-1.0.0/bin/connect-standalone.sh /path/to/your/connect-standalone.properties /path/to/your/connect-cloudant-source.properties /path/to/your/connect-jdbc-sink.properties
```

Comment:
Indent line to align with step above.

Reply:
Done in da5794d3203153f7a21961733baf16fd89702268